src/rideables/Stack.hpp

```diff
@@ -0,0 +1,71 @@
+// #ifndef STACK_HPP
+// #define STACK_HPP
+
+#include <stdio.h>
+#include<iostream>
+#include <atomic>
+
+using namespace std;
+
+class Stack
+{
+    private:
+
+    typedef struct StackNode
+    {
+
+        int data;
+        StackNode *next;
+
+    } * StackNodePtr;
+
+    atomic<StackNodePtr> top;
+
+    public:
+    Stack()
```

Comment:
You may find C++ initializer list useful for `atomic` members. See link [here](https://en.cppreference.com/w/cpp/language/constructor). So the code may be like:

```c++
Stack() : top(nullptr) {}
```